scripts/imgtool/image.py

```diff
@@ -198,8 +198,9 @@ def check(self):
         # If there is a header requested, make sure that the image
         # starts with all zeros.
         if self.header_size > 0:
-            if any(v != 0 for v in self.payload[0:self.header_size]):
-                raise Exception("Padding requested, but image does not start with zeros")
+            val = self.erased_val if self.pad_header else 0
```

Comment:
Is there any reason to even check this if pad_header is added, since we just added that header earlier?

Reply:
Yeah, no, fixed.